gatsby-node.js

```diff
@@ -94,6 +95,28 @@ exports.sourceNodes = async ({
     const children = [];
     events.forEach(event => {
       const eventNodeId = createNodeId(`${EVENT_NODE_TYPE}-${event.id}`);
+
+      // If no events have a description, then Google does not send the field.
+			// An empty value should be provided in-case queries are referencing it.
+			if(!event.description) event.description = '';
+
+			// The date field is named `date` instead of `dateTime` for all-day events.
+			// We have to translate this into the expected schema.
+			if(event.start.date && event.start.date) {
+				event.start = {
+					dateTime: moment(event.start.date).tz(calendar.timeZone).format(),
+					timeZone: calendar.timeZone
+				};
+				event.end = {
+					dateTime: moment(event.end.date).tz(calendar.timeZone).format(),
+					timeZone: calendar.timeZone
+				};
+				event.allDay = true;
+			}
+			else {
+				event.allDay = false;
+			}
```

Comment:
Although I like the idea of having a consistent event schema, I am not too sure it is a good idea to deviate from the [event schema returned from the Calendar API](https://developers.google.com/calendar/v3/reference/events#resource). In particular, removing the `*.date` fields from the event, could break existing sites that rely on these fields.

What do you think about just adding the `allDay` flag to the event? While this would require a bit more work in the site itself (checking the `allDay` flag and using either `dateTime` or `date`) it would keep changes to a minimum and consistent with the Google event schema.

Something like the following:
```
event.allDay = event.start.date && event.end.date
```

Reply:
@msigwart That would suffice just fine, but the problem here is that Gatsby totally freaks out when some events have the field named `dateTime` and others have `date`. When that happens, Gatsby only returns you the ones with the `date` field and not the `dateTime` ones.

I think Gatsby is expecting all objects to match the same schema, so it's just choosing one or the other and only showing the records for the schema it picked.

I've never made a Gatsby source plugin before so I could be wrong but that's what I'm seeing from my tests.

Reply:
Oh okay, I wasn't aware of that. Let me try it out and get back to you.

Reply:
I checked with an example project and it looks like Gatsby is able to recognise both fields. If the event is "all-day" only field `date` is populated and field `dateTime` is `null`. If the event is **not** "all-day", only field `dateTime` is populated and `date` is `null` (see screenshot below).
<img width="583" alt="Screenshot 2020-12-21 at 20 12 19" src="https://user-images.githubusercontent.com/9678787/102813815-c951c980-43c9-11eb-975c-bb8657577535.png">

Reply:
@msigwart That's very strange. Perhaps it has something to do with the other events in the calendar? There's also repeating events in there too, so not sure if that has something to do with it.

All I know is that the way it is, I only see all-day events if there's at least one of them. After these changes I've made it works perfectly.

Reply:
@chrispytoes Might be the case! Do you maybe have a public test calendar where I can replicate the issue?

Reply:
@msigwart Actually, now that I go back and look, maybe that was caused by something else. I am now seeing what you're seeing where the date fields are null when not provided. So I guess this will work for now.

But this still poses an issue:
I need to filter my events by month, so my query looks like this:
```
allCalendarEvent(
	filter: {
		start: {
			dateTime: { gte: $start }
		}
		end: {
			dateTime: { lte: $end }
		}
	}
) { ... }
```

I really can't filter for events that have `dateTime` OR `date` within the specified range. I have to make two separate queries to get all the events for the month. This is why I think having a consistent schema is better for this, because GraphQL isn't really made to handle this situation.

Reply:
@chrispytoes That is true. I guess one way to handle this could be to just set `dateTime = date` if `date` is specified. Then it would be possible to filter all events in a single query. With the additional `allDay` flag, there also wouldn't be any confusion about what can be expected in the respective fields. What do you think about that?

Reply:
@msigwart Exactly what I was thinking, but now here's the issue with that: The `date` field just gives back a `YYYY-MM-DD` date with no timezone. When I do that, all events are pulled back one day because it zeros out the timezone. That's why I needed `moment-timezone` to bring the timezone back to what the calendar had.

Reply:
Oh right, that makes sense. I'm usually not a fan of adding an external dependency for just a small code snippet but I guess this is the simplest solution right now. I'd still keep the original `date` field in the event data though so that existing sites don't break.

So I guess we're looking at something like:
```
if(event.start.date && event.end.date) {
  // event is "all-day"
  event.start = {
    ...event.start,
    dateTime: moment(event.start.date).tz(calendar.timeZone).format(),
    timeZone: calendar.timeZone
  };
  event.end = {
    ...event.end,
    dateTime: moment(event.end.date).tz(calendar.timeZone).format(),
    timeZone: calendar.timeZone
  };
  event.allDay = true;
} 
else {
  event.allDay = false;
}
```

Reply:
@msigwart Looks good to me! I've pushed that up now.